web/html/src/manager/content-management/shared/components/messages/messages.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import * as React from 'react';
+import _isEmpty from "lodash/isEmpty";
+
+import {Messages} from 'components/messages';
+import type {ProjectMessageType} from '../../type/project.type';
+
+type ValidationMessagesType = {
+  panelClass: string,
+  messages: ?React.Element<'ul'>
+}
+
+const msgClassMap = {
+  'warning': {
+    text: 'text-warning',
+    panel: 'panel-warning'
+  },
+  'error': {
+    text: 'text-danger',
+    panel: 'panel-danger'
+  },
+  'info': {
+    text: 'text-default',
+    panel: 'panel-default'
+  }
+}
+
+const sortMessages = (messages: Array<ProjectMessageType>) => {
+  const msgPriorities = {'error': 0, 'warning': 1, 'info': 2};
+  return messages.slice().sort((a, b) => msgPriorities[a.type] - msgPriorities[b.type]);
+}
+
+const getMessages = (messages: Array<ProjectMessageType>): ValidationMessagesType => {
+  if (_isEmpty(messages)) {
+    return {
+      panelClass: 'panel-default',
+      messages: null
+    }
+  }
+
+  const sortedMsgs = sortMessages(messages);
+  return {
+    panelClass: msgClassMap[sortedMsgs[0].type].panel,
+    messages: (
+      <Messages items={sortedMsgs.map(m => ({severity: m.type, text: m.text}))}/>
+    )
+  }
+}
+
+export default getMessages;
```

Comment:
2nd thing: to me it looks like this is CLM specific. Wouldn't it make sense to move this somewhere "closer" to the CLM code (`shared/components/messages`), if there is a better directory for that?

Reply:
I'm not sure I understand this one. It's already in content management domain (`manager/content-management/shared/..`). Maybe you confused it with the generic `src/components/messages.js`?

Reply:
Ah, my fault. For some reason I didn't notice it's under `content-management`. Sorry for the confusion!